.github/workflows/component.yml

```diff
@@ -55,15 +55,16 @@ jobs:
       PYTHON_BUILDER_IMAGE: builder-images/python-builder:v0.1
       GO_BUILDER_IMAGE: builder-images/go-builder:v0.1
     steps:
-      - name: Free space on runner (delete unnecessary tools)
-        run: rm -rf /opt/hostedtoolcache
-
       - name: Checkout code
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.2
         with:
           persist-credentials: false
           ref: ${{ inputs.ref || '' }}
 
+      - name: Initial cleanup
```

Comment:
If I understand correctly we are cleaning it cause there is not enough space for `interactive_ai/workflows/otx_domain/trainer/otx_v2/xpu`.  Wouldn't it be better to add if statement and check if what we are running is `interactive_ai/workflows/otx_domain/trainer/otx_v2/xpu` and if not don't do it for other components? Cause this step takes from `2X seconds` to `1,2X minute` to be executed.

Reply:
Given that the entire workflow takes 22 minutes 28 seconds, those 22 seconds amount to roughly 1 % of the total runtime, hardly excessive. Enabling this cleanup only for certain components would just clutter the workflow definition. Let’s keep it on for all components, and if it starts to take noticeably longer we can optimise it then.

Reply:
Which workflow is taking 22 minutes? Most of the individual components take ~3 minutes to complete. Even 22 seconds is then quite a significant amount. I agree with Maciej that it would be nice to only run this step for disk space restricted workflows.

Reply:
@mramotowski  @AlbertvanHouten Thanks, I will create a separate PR for that after merging this one to avoid bloating its scope.